src/root/NetworkAgnostic.ts

```diff
@@ -0,0 +1,116 @@
+import BN from 'bn.js'
+import Contract from 'web3/eth/contract'
+// import RootChainManagerArtifact from 'matic-pos-portal/artifacts/RootChainManager.json'
+import ChildTokenArtifact from 'matic-pos-portal/artifacts/ChildToken.json'
+
+import ContractsBase from '../common/ContractsBase'
+import { address } from '../types/Common'
+import Web3Client from '../common/Web3Client'
+
+// let sigUtil = require('eth-sig-util')
+
+const domainType = [
+  { name: 'name', type: 'string' },
+  { name: 'version', type: 'string' },
+  { name: 'chainId', type: 'uint256' },
+  { name: 'verifyingContract', type: 'address' },
+]
+
+const metaTransactionType = [
+  { name: 'nonce', type: 'uint256' },
+  { name: 'from', type: 'address' },
+  { name: 'functionSignature', type: 'bytes' },
+]
+
+export default class NetworkAgnostic extends ContractsBase {
+  web3Client: Web3Client
+
+  constructor(web3Client: Web3Client) {
+    super(web3Client)
+    this.web3Client = web3Client
+  }
```

Comment:
No need to set `web3Client` in this class since it is already being done in the super class.
It will be inherited here.

Reply:
got it.